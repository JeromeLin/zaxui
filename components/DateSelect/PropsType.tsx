export default interface PropsType {
  placeholder?: string;
  title?: any;
  cancelText?: string;
  okText?: string;
  mode?: 'year' | 'month' | 'date' | 'time' | 'datetime';
  disabled?: boolean;
  value?: string | object;
  defaultValue?: string | object;
  wheelDefaultValue?: string | object;
  onCancel?: () => void;
  onMaskClick?: () => void;
  onClick: () => void;
  onChange?: (value?: Date) => void;
  minuteStep?: number;
  prefixCls?: string;
  min?: Date | string;
  max?: Date | string;
  formatYear?: (num?: number) => string;
  formatMonth?: (num?: number) => string;
  formatDay?: (num?: number) => string;
  formatHour?: (num?: number) => string;
  formatMinute?: (num?: number) => string;
  locale: any;
  valueMember?: string;
}
