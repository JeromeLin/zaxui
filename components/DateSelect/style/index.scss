@import "../../style/core/index";
@import "../../style/variables";
@import "./component";