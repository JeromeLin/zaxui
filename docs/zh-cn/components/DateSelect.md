# 选择器 DateSelect

[demo页面](https://zhongantecheng.github.io/zarm/#/dateSelect)

### 引入

```js
import { DateSelect } from 'zarm';
```

### 代码演示

#### 基本用法


###### 年份选择
```jsx
<DateSelect
  title="选择年份"
  placeholder="请选择年份"
  mode="year"
  value={year}
  onChange={(value) => {
    this.setState({
      year: value,
    });
  }}
  />
```


###### 日期选择
```jsx
<DateSelect
  title="选择日期"
  placeholder="请选择日期"
  mode="date"
  value={date}
  min="2007-01-03"
  max="2017-11-23"
  onChange={(value) => {
    this.setState({
      date: value,
    });
  }}
  />
```

###### 时间选择
```jsx
<DateSelect
  title="选择时间"
  placeholder="请选择时间"
  mode="time"
  value={time}
  minuteStep={15}
  onChange={(value) => {
    this.setState({
      time: value,
    });
  }}
  />
```

###### 日期和时间选择
```jsx
<DateSelect
  title="选择日期"
  placeholder="请选择日期"
  mode="datetime"
  value={datetime}
  onChange={(value) => {
    this.setState({
      datetime: value,
    });
  }}
  />
```
###### 自定义格式日期
```jsx
<DateSelect
  title="选择日期"
  placeholder="请选择日期"
  mode="datetime"
  format="yyyy年MM月dd日 HH时mm分"
  value={customDate}
  onChange={(value) => {
    this.setState({
      customDate: value,
    });
  }}
  />
```


### API

#### DateSelect

| 属性 | 类型 | 默认值 | 可选值／参数 | 说明 |
| :--- | :--- | :--- | :--- | :--- |
| prefixCls | string | za-select | | 类名前缀 |
| className | string | | | 追加类名 |
| visible | boolean | false | | 是否展示 |
| placeholder | string | '请选择' | | 输入提示信息 |
| title | string | '请选择' | | 选择器标题 |
| cancelText | string | '取消' | | 取消栏文字 |
| okText | string | '确定' | | 确定栏文字 |
| dataSource | array | [ ] | | 数据源 |
| mode | string | date | `year`, `month`, `date`, `time`, `datetime` | 指定日期选择模式 |
| value | string |  | | 值 |
| defaultValue | string &#124; Date | | | 初始值（例：'2017-12-10' 或 new Date()） |
| format | string | | 例：YYYY年MM月DD日<br /> 年:`YYYY`, 月:`MM`, 日:`DD`, 时:`hh`, 分:`mm`。| 格式化显示值 |
| min | string | | | 相应mode的最小时间 |
| max | string | | | 相应mode的最大时间 |
| minuteStep | number | 1 | | 分钟间隔 |
| wheelDefaultValue | string | | | 滚轮默认值 |
| disabled | boolean | false | | 是否禁用 |
| onChange | <code>(value?: object) => void</code> | noop | \(value: object\) | 点击确定时触发的回调函数 |
| onCancel | <code>() => void</code> | noop | | 点击取消时触发的回调函数 |
| onMaskClick | <code>() => void</code> | noop | | 点击遮罩层时触发的回调函数 |
