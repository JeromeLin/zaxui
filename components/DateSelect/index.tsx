import React, { Component } from 'react';
import classnames from 'classnames';
import PropsType from './PropsType';
import formatFn from '../DatePickerView/utils';
import DatePicker from '../DatePicker';

function isExtendDate(date) {
  if (date instanceof Date) {
    return date;
  }

  if (!date) {
    return '';
  }

  return new Date(date.toString().replace(/-/g, '/'));
}

export interface DateSelectProps extends PropsType {
  prefixCls?: string;
  className?: any;
}

const DATE = 'date';

export default class DateSelect extends Component<DateSelectProps, any> {

  static defaultProps = {
    placeholder: '请选择',
    title: '请选择',
    cancelText: '取消',
    okText: '确定',
    mode: DATE,
    disabled: false,
    value: '',
    defaultValue: '',
    minuteStep: 1,
    prefixCls: 'za-select',
    valueMember: 'value',
    onClick: () => {},
    onCancel: () => {},
  };

  constructor(props) {
    super(props);

    const date = props.value && isExtendDate(props.value);
    const defaultDate = props.defaultValue && isExtendDate(props.defaultValue);

    this.state = {
      visible: props.visible || false,
      date: defaultDate || date,
      value: props.defaultValue || props.value || '',
    };

  }

  componentWillReceiveProps(nextProps) {
    if ('value' in nextProps) {
      const date = nextProps.value && isExtendDate(nextProps.value);
      const defaultDate = nextProps.defaultValue && isExtendDate(nextProps.defaultValue);

      this.setState({
        date: date || defaultDate,
        value: nextProps.value,
      });
    }
  }

  handleClick = () => {
    this.props.onClick();
    if (!this.props.disabled) {
      this.toggle();
    }
  }

  // 切换显示状态
  toggle() {
    this.setState({
      visible: !this.state.visible,
    });
  }

  onChange = (newValue) => {
    const { onChange } = this.props;
    if (typeof onChange === 'function') {
      onChange(newValue);
    }
    this.setState({ visible: false });
  }

  onCancel = () => {
    const { onCancel } = this.props;
    this.setState({ visible: false });
    if (typeof onCancel === 'function') {
      onCancel();
    }
  }

  render() {
    const { prefixCls, className, title, okText, cancelText, placeholder, disabled,
      onChange, ...others } = this.props;
    const { value } = this.state;
    const inputCls = classnames({
      [`${prefixCls}-input`]: true,
      [`${prefixCls}-placeholder`]: !this.state.date,
      [`${prefixCls}-disabled`]: !!disabled,
    });

    return(
      <div className={inputCls} onClick={this.handleClick}>
        <input type="hidden" value={formatFn(this, this.state.date)} />
        {this.state.date ? formatFn(this, this.state.date) : placeholder}
        <DatePicker
          visible={this.state.visible}
          onOk={this.onChange}
          value={value}
          {...others}
          onCancel={this.onCancel}
        />
      </div>
    );
  }
}
