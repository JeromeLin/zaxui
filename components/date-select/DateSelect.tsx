import React, { PureComponent } from 'react';
import classnames from 'classnames';
import { BaseDateSelectProps } from './PropsType';
import formatFn from '../date-picker-view/utils';
import DatePicker from '../date-picker';

const isExtendDate = (date) => {
  if (date instanceof Date) {
    return date;
  }

  if (!date) {
    return '';
  }

  return new Date(date.toString().replace(/-/g, '/'));
};

export interface DateSelectProps extends BaseDateSelectProps {
  prefixCls?: string;
  className?: any;
}

export default class DateSelect extends PureComponent<DateSelectProps, any> {
  static defaultProps = {
    mode: 'date',
    disabled: false,
    minuteStep: 1,
    prefixCls: 'za-date-select',
    valueMember: 'value',
    onCancel: () => {},
  };

  constructor(props) {
    super(props);

    const date = props.value && isExtendDate(props.value);
    const defaultDate = props.defaultValue && isExtendDate(props.defaultValue);

    this.state = {
      value: defaultDate || date,
      visible: props.visible || false,
    };
  }

  componentWillReceiveProps(nextProps) {
    const date = nextProps.value && isExtendDate(nextProps.value);
    const defaultDate = nextProps.defaultValue && isExtendDate(nextProps.defaultValue);

    this.setState({
      value: date || defaultDate,
    });

    if ('visible' in nextProps && nextProps.visible !== this.state.visible) {
      this.setState({
        visible: nextProps.visible,
      });
    }
  }

  handleClick = () => {
    const { disabled } = this.props;
    if (disabled) {
      return false;
    }
    this.setState({
      visible: true,
    });
  }

  onChange = (selected) => {
    const { onChange } = this.props;
    if (typeof onChange === 'function') {
      onChange(selected);
    }
  }

  onOk = (selected) => {
    const { onOk } = this.props;
    this.setState({ visible: false });
    if (typeof onOk === 'function') {
      onOk(selected);
    }
  }

  onCancel = () => {
    const { onCancel } = this.props;
    this.setState({ visible: false });
    if (typeof onCancel === 'function') {
      onCancel();
    }
  }

  render() {
    const { prefixCls, className, placeholder, disabled, onChange, locale, ...others } = this.props;
    const { value, visible } = this.state;

    const cls = classnames(prefixCls, className, {
      [`${prefixCls}--placeholder`]: !this.state.value,
      [`${prefixCls}--disabled`]: disabled,
    });

    return(
      <div className={cls} onClick={this.handleClick}>
        <input type="hidden" value={formatFn(this, this.state.value)} />
        <div className={`${prefixCls}__input`}>
          {this.state.value ? formatFn(this, this.state.value) : placeholder || locale!.placeholder}
        </div>
        <DatePicker
          {...others}
          visible={visible}
          value={value}
          onOk={this.onOk}
          onCancel={this.onCancel}
        />
      </div>
    );
  }
}
