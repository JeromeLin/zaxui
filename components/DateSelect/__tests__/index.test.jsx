import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import moment from 'moment-timezone';
import DateSelect from '../index';
import enLocale from '../../DatePicker/locale/en_US';

describe('DateSelect', () => {
  moment.tz.setDefault('Asia/Shanghai');
  it('DateSelect year', () => {
    const wrapper = mount(
      <DateSelect
        title="选择年份"
        placeholder="请选择年份"
        mode="year"
        value={moment('2017')}
        locale={enLocale}
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('DateSelect trigger visible', () => {
    const wrapper = mount(
      <DateSelect
        mode="date"
        defaultValue={moment('2017-11-03')}
        visible
        />
      );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('DateSelect month', () => {
    const wrapper = mount(
      <DateSelect
        title="选择年份"
        placeholder="请选择年份"
        defaultValue={moment('2017')}
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('DateSelect date', () => {
    const wrapper = mount(
      <DateSelect
        title="选择日期"
        placeholder="请选择日期"
        mode="date"
        value={moment('2009-03-04')}
        />
    );
    wrapper.setProps({ value: moment('2017-09-06') });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('DateSelect time', () => {
    const wrapper = mount(
      <DateSelect
        title="选择时间"
        placeholder="请选择时间"
        mode="time"
        defaultValue={moment('2017-11-03 15:00')}
        minuteStep={15}
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('DateSelect datetime', () => {
    const wrapper = mount(
      <DateSelect
        title="选择时间"
        placeholder="请选择时间"
        mode="datetime"
        min={moment('2017-11-02 11:00')}
        max={moment('2017-11-12 14:00')}
        defaultValue={moment('2017-11-03 15:00')}
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    jest.useFakeTimers();
    wrapper.setProps({ defaultValue: moment('2017-11-06 12:00'), value: moment('2017-11-06 12:00') });
    jest.runAllTimers();
  });

  it('DateSelect wheelDefaultValue', () => {
    const wrapper = mount(
      <DateSelect
        title="选择日期"
        placeholder="请选择日期"
        mode="date"
        visible
        wheelDefaultValue={moment('2017-11-03 15:00')}
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should trigger onOk when press ok button', () => {
    const onOkFn = jest.fn();
    const onCancelFn = jest.fn();

    const wrapper = mount(
      <DateSelect
        title="选择时间"
        placeholder="请选择时间"
        mode="datetime"
        defaultValue={moment('2017-11-12 20:00')}
        onOk={onOkFn}
        onCancel={onCancelFn}
        onMaskClick={onCancelFn}
        />
    );

    // wrapper.find('.za-picker-submit').simulate('click');
    // expect(onOkFn).toBeCalled();
    // expect(onCancelFn).not.toBeCalled();
  });

  it('should trigger onCancel when press cancel button', () => {
    const onOkFn = jest.fn();
    const onCancelFn = jest.fn();

    const wrapper = mount(
      <DateSelect
        title="选择时间"
        placeholder="请选择时间"
        mode="datetime"
        onOk={onOkFn}
        onCancel={onCancelFn}
        onMaskClick={onCancelFn}
        defaultValue={moment('2017-11-12 20:00')}
        />
    );

    // wrapper.find('.za-picker-cancel').simulate('click');
    // expect(onCancelFn).toBeCalled();
    // expect(onOkFn).not.toBeCalled();
  });
});
