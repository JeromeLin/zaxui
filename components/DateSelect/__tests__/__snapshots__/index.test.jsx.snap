// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateSelect DateSelect date 1`] = `
<DateSelect
  cancelText="取消"
  defaultValue=""
  disabled={false}
  minuteStep={1}
  mode="date"
  okText="确定"
  onCancel={[Function]}
  onClick={[Function]}
  placeholder="请选择日期"
  prefixCls="za-select"
  title="选择日期"
  value="2017-09-06"
  valueMember="value"
>
  <div
    className="za-select-input"
    onClick={[Function]}
  >
    <input
      type="hidden"
      value="2017-09-06"
    />
    2017-09-06
    <DatePicker
      cancelText="取消"
      defaultValue=""
      disabled={false}
      locale={
        Object {
          "day": "日",
          "hour": "时",
          "minute": "分",
          "month": "月",
          "year": "年",
        }
      }
      minuteStep={1}
      mode="date"
      okText="确定"
      onCancel={[Function]}
      onClick={[Function]}
      onOk={[Function]}
      placeholder="请选择"
      prefixCls="za-picker"
      title="请选择"
      value="2017-09-06"
      valueMember="value"
      visible={false}
    >
      <div
        className="za-picker"
      >
        <div
          className="za-picker-container za-picker-hidden"
          onClick={[Function]}
        >
          <Popup
            animationDuration={200}
            autoClose={false}
            direction="bottom"
            mask={true}
            maskType="normal"
            onMaskClick={[Function]}
            prefixCls="za-popup"
            stayTime={3000}
            visible={false}
          >
            <div
              className="za-popup za-popup-hidden"
            >
              <div
                className="za-popup-wrapper za-popup-wrapper-bottom"
                style={
                  Object {
                    "WebkitTransitionDuration": "200ms",
                    "transitionDuration": "200ms",
                  }
                }
              >
                <div
                  className="za-picker-wrapper"
                >
                  <div
                    className="za-picker-header"
                  >
                    <div
                      className="za-picker-cancel"
                      onClick={[Function]}
                    >
                      取消
                    </div>
                    <div
                      className="za-picker-title"
                    >
                      请选择
                    </div>
                    <div
                      className="za-picker-submit"
                      onClick={[Function]}
                    >
                      确定
                    </div>
                  </div>
                  <div
                    className="za-picker-mask-top"
                  >
                    <div
                      className="za-picker-mask-bottom"
                    >
                      <DatePickerView
                        cancelText="取消"
                        defaultValue=""
                        disabled={false}
                        locale={
                          Object {
                            "day": "日",
                            "hour": "时",
                            "minute": "分",
                            "month": "月",
                            "year": "年",
                          }
                        }
                        minuteStep={1}
                        mode="date"
                        okText="确定"
                        onCancel={[Function]}
                        onClick={[Function]}
                        onOk={[Function]}
                        onValueChange={[Function]}
                        placeholder="请选择"
                        prefixCls="za-picker"
                        title="请选择"
                        value="2017-09-06"
                        valueMember="value"
                        visible={false}
                      />
                    </div>
                  </div>
                </div>
              </div>
              <Mask
                className="fade-leave"
                onClose={[Function]}
                prefixCls="za-mask"
                style={
                  Object {
                    "WebkitAnimationDuration": "200ms",
                    "animationDuration": "200ms",
                  }
                }
                type="normal"
                visible={false}
              />
            </div>
          </Popup>
        </div>
      </div>
    </DatePicker>
  </div>
</DateSelect>
`;

exports[`DateSelect DateSelect datetime 1`] = `
<DateSelect
  cancelText="取消"
  defaultValue="2017-11-03 15:00"
  disabled={false}
  max="2017-11-12 14:00"
  min="2017-11-02 11:00"
  minuteStep={1}
  mode="datetime"
  okText="确定"
  onCancel={[Function]}
  onClick={[Function]}
  placeholder="请选择时间"
  prefixCls="za-select"
  title="选择时间"
  value=""
  valueMember="value"
>
  <div
    className="za-select-input"
    onClick={[Function]}
  >
    <input
      type="hidden"
      value="2017-11-03 15:00"
    />
    2017-11-03 15:00
    <DatePicker
      cancelText="取消"
      defaultValue="2017-11-03 15:00"
      disabled={false}
      locale={
        Object {
          "day": "日",
          "hour": "时",
          "minute": "分",
          "month": "月",
          "year": "年",
        }
      }
      max="2017-11-12 14:00"
      min="2017-11-02 11:00"
      minuteStep={1}
      mode="datetime"
      okText="确定"
      onCancel={[Function]}
      onClick={[Function]}
      onOk={[Function]}
      placeholder="请选择"
      prefixCls="za-picker"
      title="请选择"
      value=""
      valueMember="value"
      visible={false}
    >
      <div
        className="za-picker"
      >
        <div
          className="za-picker-container za-picker-hidden"
          onClick={[Function]}
        >
          <Popup
            animationDuration={200}
            autoClose={false}
            direction="bottom"
            mask={true}
            maskType="normal"
            onMaskClick={[Function]}
            prefixCls="za-popup"
            stayTime={3000}
            visible={false}
          >
            <div
              className="za-popup za-popup-hidden"
            >
              <div
                className="za-popup-wrapper za-popup-wrapper-bottom"
                style={
                  Object {
                    "WebkitTransitionDuration": "200ms",
                    "transitionDuration": "200ms",
                  }
                }
              >
                <div
                  className="za-picker-wrapper"
                >
                  <div
                    className="za-picker-header"
                  >
                    <div
                      className="za-picker-cancel"
                      onClick={[Function]}
                    >
                      取消
                    </div>
                    <div
                      className="za-picker-title"
                    >
                      请选择
                    </div>
                    <div
                      className="za-picker-submit"
                      onClick={[Function]}
                    >
                      确定
                    </div>
                  </div>
                  <div
                    className="za-picker-mask-top"
                  >
                    <div
                      className="za-picker-mask-bottom"
                    >
                      <DatePickerView
                        cancelText="取消"
                        defaultValue="2017-11-03 15:00"
                        disabled={false}
                        locale={
                          Object {
                            "day": "日",
                            "hour": "时",
                            "minute": "分",
                            "month": "月",
                            "year": "年",
                          }
                        }
                        max="2017-11-12 14:00"
                        min="2017-11-02 11:00"
                        minuteStep={1}
                        mode="datetime"
                        okText="确定"
                        onCancel={[Function]}
                        onClick={[Function]}
                        onOk={[Function]}
                        onValueChange={[Function]}
                        placeholder="请选择"
                        prefixCls="za-picker"
                        title="请选择"
                        value="2017-11-03 15:00"
                        valueMember="value"
                        visible={false}
                      />
                    </div>
                  </div>
                </div>
              </div>
              <Mask
                className=""
                onClose={[Function]}
                prefixCls="za-mask"
                style={
                  Object {
                    "WebkitAnimationDuration": "200ms",
                    "animationDuration": "200ms",
                  }
                }
                type="normal"
                visible={false}
              />
            </div>
          </Popup>
        </div>
      </div>
    </DatePicker>
  </div>
</DateSelect>
`;

exports[`DateSelect DateSelect month 1`] = `
<div
  class="za-select-input za-select-placeholder"
>
  <input
    type="hidden"
    value=""
  />
  请选择年份
  <div
    class="za-picker"
  >
    <div
      class="za-picker-container za-picker-hidden"
    >
      <div
        class="za-popup za-popup-hidden"
      >
        <div
          class="za-popup-wrapper za-popup-wrapper-bottom"
          style="-webkit-transition-duration:200ms;transition-duration:200ms"
        >
          <div
            class="za-picker-wrapper"
          >
            <div
              class="za-picker-header"
            >
              <div
                class="za-picker-cancel"
              >
                取消
              </div>
              <div
                class="za-picker-title"
              >
                请选择
              </div>
              <div
                class="za-picker-submit"
              >
                确定
              </div>
            </div>
            <div
              class="za-picker-mask-top"
            >
              <div
                class="za-picker-mask-bottom"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DateSelect DateSelect time 1`] = `
<DateSelect
  cancelText="取消"
  defaultValue="2017-11-03 15:00"
  disabled={false}
  minuteStep={15}
  mode="time"
  okText="确定"
  onCancel={[Function]}
  onClick={[Function]}
  placeholder="请选择时间"
  prefixCls="za-select"
  title="选择时间"
  value=""
  valueMember="value"
>
  <div
    className="za-select-input"
    onClick={[Function]}
  >
    <input
      type="hidden"
      value="15:00"
    />
    15:00
    <DatePicker
      cancelText="取消"
      defaultValue="2017-11-03 15:00"
      disabled={false}
      locale={
        Object {
          "day": "日",
          "hour": "时",
          "minute": "分",
          "month": "月",
          "year": "年",
        }
      }
      minuteStep={15}
      mode="time"
      okText="确定"
      onCancel={[Function]}
      onClick={[Function]}
      onOk={[Function]}
      placeholder="请选择"
      prefixCls="za-picker"
      title="请选择"
      value=""
      valueMember="value"
      visible={false}
    >
      <div
        className="za-picker"
      >
        <div
          className="za-picker-container za-picker-hidden"
          onClick={[Function]}
        >
          <Popup
            animationDuration={200}
            autoClose={false}
            direction="bottom"
            mask={true}
            maskType="normal"
            onMaskClick={[Function]}
            prefixCls="za-popup"
            stayTime={3000}
            visible={false}
          >
            <div
              className="za-popup za-popup-hidden"
            >
              <div
                className="za-popup-wrapper za-popup-wrapper-bottom"
                style={
                  Object {
                    "WebkitTransitionDuration": "200ms",
                    "transitionDuration": "200ms",
                  }
                }
              >
                <div
                  className="za-picker-wrapper"
                >
                  <div
                    className="za-picker-header"
                  >
                    <div
                      className="za-picker-cancel"
                      onClick={[Function]}
                    >
                      取消
                    </div>
                    <div
                      className="za-picker-title"
                    >
                      请选择
                    </div>
                    <div
                      className="za-picker-submit"
                      onClick={[Function]}
                    >
                      确定
                    </div>
                  </div>
                  <div
                    className="za-picker-mask-top"
                  >
                    <div
                      className="za-picker-mask-bottom"
                    >
                      <DatePickerView
                        cancelText="取消"
                        defaultValue="2017-11-03 15:00"
                        disabled={false}
                        locale={
                          Object {
                            "day": "日",
                            "hour": "时",
                            "minute": "分",
                            "month": "月",
                            "year": "年",
                          }
                        }
                        minuteStep={15}
                        mode="time"
                        okText="确定"
                        onCancel={[Function]}
                        onClick={[Function]}
                        onOk={[Function]}
                        onValueChange={[Function]}
                        placeholder="请选择"
                        prefixCls="za-picker"
                        title="请选择"
                        value="2017-11-03 15:00"
                        valueMember="value"
                        visible={false}
                      />
                    </div>
                  </div>
                </div>
              </div>
              <Mask
                className=""
                onClose={[Function]}
                prefixCls="za-mask"
                style={
                  Object {
                    "WebkitAnimationDuration": "200ms",
                    "animationDuration": "200ms",
                  }
                }
                type="normal"
                visible={false}
              />
            </div>
          </Popup>
        </div>
      </div>
    </DatePicker>
  </div>
</DateSelect>
`;

exports[`DateSelect DateSelect year 1`] = `
<div
  class="za-select-input"
>
  <input
    type="hidden"
    value="2017年"
  />
  2017年
  <div
    class="za-picker"
  >
    <div
      class="za-picker-container za-picker-hidden"
    >
      <div
        class="za-popup za-popup-hidden"
      >
        <div
          class="za-popup-wrapper za-popup-wrapper-bottom"
          style="-webkit-transition-duration:200ms;transition-duration:200ms"
        >
          <div
            class="za-picker-wrapper"
          >
            <div
              class="za-picker-header"
            >
              <div
                class="za-picker-cancel"
              >
                取消
              </div>
              <div
                class="za-picker-title"
              >
                请选择
              </div>
              <div
                class="za-picker-submit"
              >
                确定
              </div>
            </div>
            <div
              class="za-picker-mask-top"
            >
              <div
                class="za-picker-mask-bottom"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
